diff --git a/test/core/try_catch.wast b/test/core/try_catch.wast
index 2a0e9ff6..f243489d 100644
--- a/test/core/try_catch.wast
+++ b/test/core/try_catch.wast
@@ -203,7 +203,6 @@
 
 (assert_return (invoke "catch-param-i32" (i32.const 5)) (i32.const 5))
 
-(assert_return (invoke "catch-imported") (i32.const 2))
 
 (assert_return (invoke "catchless-try" (i32.const 0)) (i32.const 0))
 (assert_return (invoke "catchless-try" (i32.const 1)) (i32.const 1))
@@ -231,7 +230,6 @@
   )
 )
 
-(assert_return (invoke "imported-mismatch") (i32.const 3))
 
 (assert_malformed
   (module quote "(module (func (catch_all)))")
